src/m365/spo/commands/listitem/listitem-batch-set.ts

```diff
@@ -0,0 +1,290 @@
+          break;
+        case 'LookupMulti':
+          const lookupMultiString: string[] = row[field.InternalName].split(';').map((element: string) => {
+            return `<Object TypeId="{f1d34cc0-9b50-4a78-be78-d5facfcccfb7}"><Property Name="LookupId" Type="Int32">${element}</Property><Property Name="LookupValue" Type="Null" /></Object>`;
+          });
+          actionString += `<Method Name="SetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="Array">${lookupMultiString.join('')}</Parameter></Parameters></Method>`;
+          break;
+        default:
+          actionString += `<Method Name="ParseAndSetFieldValue" Id="${index += 1}" ObjectPathId="${objectPathId}"><Parameters><Parameter Type="String">${field.InternalName}</Parameter><Parameter Type="String">${(<any>row)[field.InternalName].toString()}</Parameter></Parameters></Method>`;
+          break;
+      }
+    });
+
+    actionString += `<Method Name="${systemUpdate ? 'System' : ''}Update" Id="${index += 1}" ObjectPathId="${objectPathId}"/>`;
+    return [actionString, index];
+  }
+
+  private async getListFields(options: Options, listId: string, jsonContent: any, idColumn: string, logger: Logger): Promise<FieldDetails[]> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving fields for list with id ${listId}`);
+    }
+
+    const filterFields: string[] = [];
+    const objectKeys = Object.keys(jsonContent[0]);
+
+    const index = objectKeys.indexOf(idColumn, 0);
+    if (index > -1) {
+      objectKeys.splice(index, 1);
+    }
+
+    objectKeys.map(objectKey => {
+      filterFields.push(`InternalName eq '${objectKey}'`);
+    });
+
+    const fields = await odata.getAllItems<FieldDetails>(`${options.webUrl}/_api/web/lists(guid'${formatting.encodeQueryParameter(listId)}')/fields?$select=InternalName,TypeAsString&$filter=${filterFields.join(' or ')}`);
+    if (fields.length !== objectKeys.length) {
+      const fieldsThatDontExist: string[] = [];
+      objectKeys.forEach(key => {
+        const field = fields.find(field => field.InternalName === key);
+        if (!field) {
+          fieldsThatDontExist.push(key);
+        }
+      });
+      throw `Following fields specified in the csv do not exist on the list: ${fieldsThatDontExist.join(', ')}`;
+    }
+    return fields;
+  }
+
+  private async getListId(options: Options, logger: Logger): Promise<string> {
+    if (this.verbose) {
+      logger.logToStderr('Retrieving list id');
+    }
+
+    let listUrl = `${options.webUrl}/_api/web`;
+    if (options.listId) {
```

Comment:
Let's return the listId immediately if we have it. So move this if-statement above the verbose logging line.